File: validate.js
#!/usr/bin/env node

/**
 * Utility script to validate configuration
 */

const fs = require('fs');
const path = require('path');
require('dotenv').config();

console.log('🔍 Validating Facebook Scraper Configuration...\n');

let hasErrors = false;

// Check for required environment variables
const requiredVars = ['FACEBOOK_ACCESS_TOKEN', 'FACEBOOK_PAGE_ID'];

requiredVars.forEach(varName => {
  if (!process.env[varName]) {
    console.error(`❌ Missing required environment variable: ${varName}`);
    hasErrors = true;
  } else {
    console.log(`✓ ${varName} is set`);
  }
});

// Check output directory
const outputDir = process.env.OUTPUT_DIR || './output';
if (!fs.existsSync(outputDir)) {
  console.log(`\n📁 Creating output directory: ${outputDir}`);
  fs.mkdirSync(outputDir, { recursive: true });
  console.log('✓ Output directory created');
} else {
  console.log(`✓ Output directory exists: ${outputDir}`);
}

// Check if dependencies are installed
console.log('\n📦 Checking dependencies...');
try {
  require('axios');
  console.log('✓ axios installed');
} catch (e) {
  console.error('❌ axios not found. Run: npm install');
  hasErrors = true;
}

try {
  require('moment');
  console.log('✓ moment installed');
} catch (e) {
  console.error('❌ moment not found. Run: npm install');
  hasErrors = true;
}

// Summary
console.log('\n' + (hasErrors ? '❌ Validation failed' : '✅ Validation successful'));
console.log(hasErrors ? '\nPlease fix the above errors before running the scraper.' : '\nYou are ready to run the scraper!');
console.log('\nUsage: node scraper.js [options]');
console.log('Run: node scraper.js --help for more options\n');

process.exit(hasErrors ? 1 : 0);

File: validate.py
#!/usr/bin/env python3

"""
Utility script to validate configuration
"""

import os
import sys
from pathlib import Path
from dotenv import load_dotenv

load_dotenv()

print('🔍 Validating Facebook Scraper Configuration...\n')

has_errors = False

# Check for required environment variables
required_vars = ['FACEBOOK_ACCESS_TOKEN', 'FACEBOOK_PAGE_ID']

for var_name in required_vars:
    if not os.getenv(var_name):
        print(f'❌ Missing required environment variable: {var_name}')
        has_errors = True
    else:
        print(f'✓ {var_name} is set')

# Check output directory
output_dir = os.getenv('OUTPUT_DIR', './output')
if not os.path.exists(output_dir):
    print(f'\n📁 Creating output directory: {output_dir}')
    Path(output_dir).mkdir(parents=True, exist_ok=True)
    print('✓ Output directory created')
else:
    print(f'✓ Output directory exists: {output_dir}')

# Check if dependencies are installed
print('\n📦 Checking dependencies...')
try:
    import requests
    print('✓ requests installed')
except ImportError:
    print('❌ requests not found. Run: pip install -r requirements.txt')
    has_errors = True

try:
    import dotenv
    print('✓ python-dotenv installed')
except ImportError:
    print('❌ python-dotenv not found. Run: pip install -r requirements.txt')
    has_errors = True

# Summary
print('\n' + ('❌ Validation failed' if has_errors else '✅ Validation successful'))
print('\nPlease fix the above errors before running the scraper.' if has_errors else '\nYou are ready to run the scraper!')
print('\nUsage: python scraper.py [options]')
print('Run: python scraper.py --help for more options\n')

sys.exit(1 if has_errors else 0)
